src/main/java/com/epam/test_generator/controllers/StepSuggestionController.java

```diff
@@ -44,6 +44,54 @@
         return new ResponseEntity<>(stepSuggestionService.getStepsSuggestions(), HttpStatus.OK);
     }
 
+    @ApiOperation(value = "Get page of step suggestions", nickname = "getStepsSuggestions")
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK", response = StepSuggestionDTO.class,
+                    responseContainer = "List")
+    })
+    @Secured({"ROLE_ADMIN", "ROLE_TEST_ENGINEER", "ROLE_TEST_LEAD", "ROLE_GUEST"})
+    @RequestMapping(value = "/stepSuggestions/page/{pageNumber}/size/{pageSize}", method = RequestMethod.GET,
+            produces = "application/json")
+    @ApiImplicitParam(name = "Authorization", value = "add here your token", paramType = "header", dataType = "string", required = true)
+    public ResponseEntity<List<StepSuggestionDTO>> getStepsSuggestions(@PathVariable int pageNumber, @PathVariable int pageSize) {
+
+        return new ResponseEntity<>(stepSuggestionService.getStepsSuggestions(pageNumber, pageSize), HttpStatus.OK);
+    }
+
+    @ApiOperation(
+            value = "Get page of step suggestions for an appropriate type",
+            nickname = "getStepsSuggestionsByType")
+    @ApiResponses(value = {
+            @ApiResponse(
+                    code = 200,
+                    message = "OK",
+                    response = StepSuggestionDTO.class,
+                    responseContainer = "List")
+    })
+    @ApiImplicitParams({
+            @ApiImplicitParam(
+                    name = "stepType",
+                    value = "Type of step suggestion that we want to return",
+                    required = true,
+                    dataType = "StepType",
+                    paramType = "path"),
+            @ApiImplicitParam(
+                    name = "Authorization",
+                    value = "add here your token",
+                    paramType = "header",
+                    dataType = "string",
+                    required = true)
+    })
+    @Secured({"ROLE_ADMIN", "ROLE_TEST_ENGINEER", "ROLE_TEST_LEAD", "ROLE_GUEST"})
+    @RequestMapping(value = "/stepSuggestions/{stepType}/page/{pageNumber}/size/{pageSize}",
```

Comment:
stepType, pageNumber and pageSize are not resources, therefore should not be declared as PathVariables. Use RequestParam for these parameters. http://www.baeldung.com/rest-api-pagination-in-spring

Reply:
Btw method getStepsSuggestionsByType follows wrong crud design pattern. I'd recommend you implement new one which takes stepType as request parameter, not path variable. Leave old method (endpoint) for backward compatibility and mark it as deprecated.